api/client/pull.go

```diff
@@ -17,24 +16,29 @@ import (
 func (cli *DockerCli) CmdPull(args ...string) error {
 	cmd := cli.Subcmd("pull", []string{"NAME[:TAG|@DIGEST]"}, "Pull an image or a repository from a registry", true)
 	allTags := cmd.Bool([]string{"a", "-all-tags"}, false, "Download all tagged images in the repository")
+	addTrustedFlags(cmd)
 	cmd.Require(flag.Exact, 1)
 
 	cmd.ParseFlags(args, true)
+	remote := cmd.Arg(0)
 
-	var (
-		v         = url.Values{}
-		remote    = cmd.Arg(0)
-		newRemote = remote
-	)
 	taglessRemote, tag := parsers.ParseRepositoryTag(remote)
 	if tag == "" && !*allTags {
-		newRemote = utils.ImageReference(taglessRemote, tags.DEFAULTTAG)
-	}
-	if tag != "" && *allTags {
+		tag = tags.DEFAULTTAG
+		fmt.Fprintf(cli.out, "Using default tag: %s\n", tag)
+	} else if tag != "" && *allTags {
 		return fmt.Errorf("tag can't be used with --all-tags/-a")
 	}
 
-	v.Set("fromImage", newRemote)
+	ref := registry.ParseReference(tag)
+
+	if isTrusted("") && !ref.IsTrusted() {
```

Comment:
this was my b i thought it was also being done in a pull file

On Mon, Jul 13, 2015 at 4:44 PM, Derek McGowan <notifications@github.com>
wrote:

> In api/client/pull.go
> <https://github.com/docker/docker/pull/14546#discussion_r34524690>:
>
> >  		return fmt.Errorf("tag can't be used with --all-tags/-a")
> >  	}
> >
> > -	v.Set("fromImage", newRemote)
> > +	ref := registry.ParseReference(tag)
> > +
> > +	if isTrusted("") && !ref.IsTrusted() {
>
> referring to which part of the statement? The !ref.IsTrusted() sends pull
> by digests down the existing code path even when trust is enabled. It could
> go the trusted path as well, would just require a check similar to create,
> only create a tag if asked for untrusted tag ubuntu:latest but pulled
> trusted ubuntu@sha256....
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/docker/docker/pull/14546/files#r34524690>.
>